spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskLifecycleListener.java

```diff
@@ -165,18 +169,34 @@ else if(this.applicationFailedEvent != null){
 			if(this.applicationFailedEvent != null) {
 				this.taskExecution.setErrorMessage(stackTraceToString(this.applicationFailedEvent.getException()));
 			}
+			//append the ApplicationFailedEvent exception with the listener errorMessage.
+			if(this.listenerException != null) {
+				String errorMessage = this.stackTraceToString(this.listenerException);
+				if (StringUtils.hasText(this.taskExecution.getErrorMessage())) {
+					errorMessage = String.format("%s :Task Also threw this Exception: %s", errorMessage, this.taskExecution.getErrorMessage());
+				}
+				this.taskExecution.setErrorMessage(errorMessage);
+			}
+
+			if(this.listenerFailed && this.taskExecution.getExitCode() != 0) {
+				setMessages(invokeOnTaskError(this.taskExecution, this.listenerException));
+			}
+			else if (this.applicationFailedEvent != null && this.taskExecution.getExitCode() != 0) {
+				setMessages(invokeOnTaskError(this.taskExecution,
+						this.applicationFailedEvent.getException()));
+			}
 
-			if (this.applicationFailedEvent != null && this.taskExecution.getExitCode() != 0) {
-				taskExecution.setExitMessage(invokeOnTaskError(taskExecution,
-						this.applicationFailedEvent.getException()).getExitMessage());
+			setMessages(invokeOnTaskEnd(this.taskExecution));
+			if(this.listenerFailed) {
+				this.taskExecution.setExitCode(1);
```

Comment:
Should this be hardcoded as 1, a specific value that we specify other than one (to indicate the specific error condition being an exception was thrown via a SCT listener), or should this be the result of any configured `ExitCodeGenerator`?  Where in the lifecycle does the `ExitCodeGenerator` get called based on this logic?

Reply:
With this PR we stop processing all TaskExecutionListeners when an exception is thrown by a listener.  So if I understand you correctly, you want to check to see if the exception thrown by the listener implements the ExitCodeGenerator interface and if so use that exit code.

Reply:
No.  I wouldn't expect an exception to implement `ExitCodeGenerator`.  However, there may be one in the `ApplicationContext`.  I'm asking if we should use it...